src/daft-local-execution/src/intermediate_ops/sample.rs

```diff
@@ -0,0 +1,47 @@
+use std::sync::Arc;
+
+use common_error::DaftResult;
+use tracing::instrument;
+
+use super::intermediate_op::{
+    IntermediateOperator, IntermediateOperatorResult, IntermediateOperatorState,
+};
+use crate::pipeline::PipelineResultType;
+
+pub struct SampleOperator {
+    fraction: f64,
+    with_replacement: bool,
+    seed: Option<u64>,
+}
+
+impl SampleOperator {
+    pub fn new(fraction: f64, with_replacement: bool, seed: Option<u64>) -> Self {
+        Self {
+            fraction,
+            with_replacement,
+            seed,
+        }
+    }
+}
+
+impl IntermediateOperator for SampleOperator {
+    #[instrument(skip_all, name = "SampleOperator::execute")]
+    fn execute(
+        &self,
+        _idx: usize,
+        input: &PipelineResultType,
+        _state: Option<&mut Box<dyn IntermediateOperatorState>>,
+    ) -> DaftResult<IntermediateOperatorResult> {
+        let out =
+            input
+                .as_data()
+                .sample_by_fraction(self.fraction, self.with_replacement, self.seed)?;
+        Ok(IntermediateOperatorResult::NeedMoreInput(Some(Arc::new(
```

Comment:
I'm not sure I follow with this. Every time we execute this operator, we're always going to request for more data?

Reply:
Instead of a 'request for more input' kind of situation, it's more like, 'this operator is ready to accept more input`. Most intermediate operators will return this as the result